plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkProvider.java

```diff
@@ -17,89 +17,40 @@
 package com.google.cloud.tools.eclipse.sdk;
 
 import com.google.cloud.tools.appengine.cloudsdk.CloudSdk;
-import com.google.cloud.tools.appengine.cloudsdk.PathResolver;
 import com.google.cloud.tools.eclipse.sdk.internal.PreferenceConstants;
 import com.google.cloud.tools.eclipse.sdk.internal.PreferenceInitializer;
 
 import org.eclipse.e4.core.contexts.ContextFunction;
 import org.eclipse.jface.preference.IPreferenceStore;
 
-import java.io.File;
-import java.nio.file.Path;
+import java.nio.file.Paths;
 
 /**
  * Utility to find the Google Cloud SDK either at locations configured by the user or in standard
  * locations on the system.
  */
 public class CloudSdkProvider extends ContextFunction {
 
-  private final IPreferenceStore preferences;
-
-   public CloudSdkProvider(IPreferenceStore preferences) {
-     // todo drop the null check and use the no args constructor instead
-     if (preferences == null) {
-       preferences = PreferenceInitializer.getPreferenceStore();
-     }
-     this.preferences = preferences;
-   }
-   
-   public CloudSdkProvider() {
-     this(PreferenceInitializer.getPreferenceStore());
-   }
+  private final IPreferenceStore preferences = PreferenceInitializer.getPreferenceStore();
 
   /**
    * Return the {@link CloudSdk} instance from the configured or discovered Cloud SDK.
    * 
    * @return the configured {@link CloudSdk} or {@code null} if no SDK could be found
    */
   public CloudSdk getCloudSdk() {
-    return createBuilder(null).build();
-  }
-
-  /**
-   * Return the location of the configured or discovered Cloud SDK.
-   * 
-   * @return the configured location or {@code null} if the SDK could not be found
-   */
-  public File getCloudSdkLocation() {
-    return resolveSdkLocation();
-  }
-
-  /**
-   * Return a {@link CloudSdk.Builder} instance, suitable for creating a {@link CloudSdk} instance.
-   * {@code location}, if not null, is used as the location to the Cloud SDK, otherwise the
-   * configured or discovered Cloud SDK will be used instead.
-   * 
-   * @param location if not {@code null}, overrides the default location for the SDK; can be a
-   *        {@linkplain String}, {@linkplain File}, or {@linkplain Path}.
-   * @return a builder, or {@code null} if the Google Cloud SDK cannot be located
-   */
-  public CloudSdk.Builder createBuilder(File location) {
```

Comment:
Yes, though since in this plugin we're using CloudSdkProvider, I think it comes out prettier if we overload. We're not likely going to need many more settings.